src/app/modules/test-navigator/test-navigator.component.spec.ts

```diff
@@ -1039,4 +1042,93 @@ describe('TestNavigatorComponent', () => {
       expect(testTclUserActivityIcon.nativeElement.title).toBeFalsy();
   })));
 
+  it('puts collaborator activities into parent nodes if the actual node does not exist',
+  fakeAsync(inject([MessagingService], async (messageBus: MessagingService) => {
+    // given
+    await component.updateModel();
+    const subfolder = component.model.children[0];
+    const collaboratorActivities: ElementActivity[] = [{
+      element: 'src/test/java/subfolder/newfolder/newfile.ext', activities: [{user: 'Jane Doe', type: 'created.file'}]
+    }];
+
+    // when
+    messageBus.publish(USER_ACTIVITY_UPDATED, collaboratorActivities);
+    tick();
+    fixture.detectChanges();
+
+    // then
+    expect(component.model.activities.getTypes('src/test/java/subfolder/newfolder/newfile.ext')).toContain('created.file');
+    expect(component.model.activities.getUsers('created.file')).toContain('Jane Doe');
+    expect(subfolder.activities.getTypes('src/test/java/subfolder/newfolder/newfile.ext')).toContain('created.file');
+    expect(subfolder.activities.getUsers('created.file', 'src/test/java/subfolder/newfolder/newfile.ext')).toContain('Jane Doe');
+  })));
+
+  it('removes collaborator activities on non-existing nodes from parent nodes when update does not contain them anymore',
+  fakeAsync(inject([MessagingService], async (messageBus: MessagingService) => {
+    // given
+    await component.updateModel();
+    const subfolder = component.model.children[0];
+    const collaboratorActivities: ElementActivity[] = [{
+      element: 'src/test/java/subfolder/newfolder/newfile.ext', activities: [{user: 'Jane Doe', type: 'created.file'}]
+    }];
+    messageBus.publish(USER_ACTIVITY_UPDATED, collaboratorActivities);
+    tick();
+
+    // when
+    messageBus.publish(USER_ACTIVITY_UPDATED, []);
+    tick();
+
```

Comment:
It should neither be necessary here, nor was it necessary in the previous test, because the expectations don't work on the DOM tree. I removed the corresponding line in the previous test.